libs/providers/flagsmith/src/lib/flagsmith-provider.ts

```diff
@@ -59,33 +108,42 @@ export default class FlagsmithOpenFeatureProvider implements Provider {
     try {
       const flags = await this.getFlags(evaluationContext);
       flag = flags.getFlag(flagKey);
-    } catch (ex) {
-      throw new FlagsmithProviderError(
-        'An error occurred retrieving flags from Flagsmith client.',
-        ErrorCode.GENERAL,
-      );
+    } catch (error) {
+      throw new FlagsmithProviderError('An error occurred retrieving flags from Flagsmith client.', ErrorCode.GENERAL);
     }
 
-    if (flag.isDefault && !this.useFlagsmithDefaults) {
+    // Do we want to raise an error if the flag is not found?
+    if (!this.useFlagsmithDefaults && (!flag || flag?.isDefault)) {
       throw new FlagNotFoundError(`Flag '${flagKey}' was not found.`);
     }
 
-    if (flagType === 'boolean' && !this.useBooleanConfigValue) {
-      return { value: flag.enabled };
+    if (!(this.returnValueForDisabledFlags || flag.enabled)) {
+      throw new FlagsmithProviderError(`Flag '${flagKey}' is not enabled.`, ErrorCode.GENERAL);
     }
 
-    if (!(this.returnValueForDisabledFlags || flag.enabled)) {
-      throw new FlagsmithProviderError(
-        `Flag '${flagKey}' is not enabled.`,
-        ErrorCode.GENERAL,
-      );
+    if (!this.useBooleanConfigValue && flagType === 'boolean') {
+      return {
+        value: flag.enabled,
+        reason: flag.enabled ? StandardResolutionReasons.TARGETING_MATCH : StandardResolutionReasons.DISABLED,
+      };
     }
 
     const typedValue = typeFactory(flag.value, flagType);
```

Comment:
typeFactory returns plain `undefined` if type mismatches and throws an error straight after. 

Did I understand well or is it too strict?